src/lib/Uploads.ts

```diff
@@ -0,0 +1,395 @@
+            ? this.config.sanitizeFilename(uploadedAs)
+            : defaultSanitizeFilename(uploadedAs);
+    }
+
+    /**
+     * Generate a timestamped unique path and filename based on the client-provided filename.
+     *
+     * @param sanitizedUploadedAs
+     */
+    public generatePath(sanitizedUploadedAs: string): string {
+        return this.config.generatePath
+            ? this.config.generatePath(sanitizedUploadedAs)
+            : defaultGeneratePath(sanitizedUploadedAs);
+    }
+
+    /**
+     * Get the full final storage path on the disk for an upload being stored based off of the sanitized filename
+     * and the configuration path prefix.
+     *
+     * @param sanitizedUploadedAs
+     */
+    public generateStoragePath(sanitizedUploadedAs: string): string {
+        const path = this.generatePath(sanitizedUploadedAs);
+        const trimmedPrefix = trimPath(`${this.config.pathPrefix || ''}`);
+        const prefix = trimmedPrefix ? `${trimmedPrefix}/` : '';
+        return `/${prefix}${trimPath(path)}`;
+    }
+
+    /**
+     * Get the default disk name based on config.
+     */
+    private getDefaultDiskName(): string {
+        return this.config.defaultDisk || 'default';
+    }
+
+    /**
+     * Take the file data and raw client-provided filename and place that file on the disk in the proper
+     * location by sanitizing the filename and generating a unique path for it.
+     *
+     * This doesn't touch the repository. Use `upload` to perform legitimate uploads.
+     *
+     * @see upload
+     *
+     * @param fileData
+     * @param uploadedAs
+     * @param diskName
+     */
+    public async place(
+        fileData: Buffer | Readable,
+        uploadedAs: string,
+        diskName: string = this.getDefaultDiskName(),
+    ): Promise<UploadedFile> {
+        // Generate a filename and path.
+        const sanitizedUploadedAs = this.sanitizeFilename(uploadedAs);
+        const path = this.generateStoragePath(sanitizedUploadedAs);
+        // Grab a disk instance and write the data to the disk.
+        const disk = this.disks.getDisk(diskName);
+        await disk.write(path, fileData);
+        // Return a record representing the uploaded file and where it lives on the disk.
+        return {
+            disk: disk.getName() || diskName,
+            path,
+            uploadedAs: sanitizedUploadedAs,
+        };
+    }
+
+    /**
+     * Take an uploaded file and copy it to a new unique generated location on the default disk (or the specified disk,
+     * regardless of the source disk). By default the path will be regenerated but this can be disabled, noting that
+     * without regeneration, it will still use the old pathPrefix and not any new one if its changed in the config
+     * (since the pathPrefix is applied during the generation of the filename).
+     *
+     * TODO Once @carimus/node-disks supports the copy operation, use that when oldDisk === newDisk
```

Comment:
Does a task need to be generated somewhere to do this or is this already noted in a task for when this will occur? (i.e. do we have a task to update @carimus/node-disks to support the copy operation and/or to make this update one it does)

Reply:
We don't have a task because there's nowhere to track tasks for these common projects yet :/ Once we set up a JIRA project for these things, I'll do the honors of digging through and finding all of these TODOs and creating tasks for them.